01-Login/README.md

```diff
@@ -1,84 +1,705 @@
+
+::: note
+You can also embed the login dialog directly in your application using the [Lock widget](/lock). If you use this method, some features, such as single sign-on, will not be accessible. 
+<% if (typeof embeddedLoginLink == 'string') { %>
+To learn how to embed the Lock widget in your application, follow the [Embedded Login sample](${embeddedLoginLink}).
+<% } %>
+:::
+
+When a user logs in, Auth0 returns three items:
+* `access_token`: to learn more, see the [Access Token documentation](/tokens/access-tokens)
+* `id_token`: to learn more, see the [ID Token documentation](/tokens/id-tokens)
+* `expires_in`: the number of seconds before the Access Token expires
+
+You can use these items in your application to set up and manage authentication. 
 
-The project needs to be configured with your Auth0 domain and client ID in order for the authentication flow to work.
+### Create an Authentication Service
 
-To do this, first copy `auth_config.sample.json` into a new file in the same folder called `auth_config.json`, and replace the values within with your own Auth0 application credentials:
+The best way to manage and coordinate the tasks necessary for user authentication is to create a reusable service. With the service in place, you will be able to call its methods throughout your application. The name for it is at your discretion, but in these examples it will be called `AuthService` and the filename will be `authService.js`. An instance of the `WebAuth` object from **auth0.js** can be created in the service.
+
+Create a service and instantiate `auth0.WebAuth`. Provide a method called `login` which calls the `authorize` method from auth0.js.
+
+```js
+// src/auth/authService.js
+
+import auth0 from 'auth0-js';
+import EventEmitter from 'events';
+import authConfig from '../../auth_config.json';
+
+const webAuth = new auth0.WebAuth({
+  domain: authConfig.domain,
+  redirectUri: `<%= "${window.location.origin}" %>/callback`,
+  clientID: authConfig.clientId,
+  responseType: 'id_token',
+  scope: 'openid profile email'
+});
+
+class AuthService extends EventEmitter {
+
+  // Starts the user login flow
+  login(customState) {
+    webAuth.authorize({
+      appState: customState
+    });
+  }
+
+}
+
+export default new AuthService();
+```
+
+:::note
+The `login` method has been setup to support specifing custom state that will be returned to the application after authentication. This will come into play later when you start adding protected routes.
+:::
+
+To provide the values for `clientID`, `callbackUrl`, and `domain`, create a new file `auth_config.json` in the root directory of the application alongside your `package.json` file, and populate it with your tenant values:
 
 ```json
 {
-  "domain": "<YOUR AUTH0 DOMAIN>",
-  "clientId": "<YOUR AUTH0 CLIENT ID>"
+  "domain": "${account.namespace}",
```

Comment:
These variables don't mean anything in GitHub. I would revert these lines to the previous version, so that the JSON example looks like this:

```json
{
  "domain": "<YOUR AUTH0 DOMAIN>",
  "clientId": "<YOUR AUTH0 CLIENT ID>"
}
```